apps/settings/js/panels/languages/languages.js

```diff
@@ -30,12 +30,28 @@ define(function(require) {
       // update the date and time samples in the 'languages' panel
       if (this.panel.children.length) {
         var d = new Date();
-        var f = new navigator.mozL10n.DateTimeFormat();
-        var _ = navigator.mozL10n.get;
-        this.panel.querySelector('#region-date').textContent =
-          f.localeFormat(d, _('longDateFormat'));
-        this.panel.querySelector('#region-time').textContent =
-          f.localeFormat(d, _('shortTimeFormat'));
+        var dateString = d.toLocaleString(navigator.languages, {
+          weekday: 'long',
+          day: 'numeric',
+          month: 'long',
+          year: 'numeric'
+        });
+        var timeString = d.toLocaleString(navigator.languages, {
+          hour12: navigator.mozHour12,
+          hour: 'numeric',
+          minute: 'numeric'
+        });
+
+        if (navigator.language in navigator.mozL10n.qps) {
```

Comment:
Yes!

So, we introduced a concept of pseudo-locales that are meant to help developers work with testing their code in different languages before localizers provide translations. Historically we had 4 languages (ar, fr, zh-CN and en-US) in our source code, but that didn't work well since the other locales were outdated and localizers don't translate at the time when developers work on code.

So pseudo-locales just take DEFAULT_LOCALE (usually en_US) and morph it to something that looks similar but different.
* qps-ploc - Is Accented English which looks like en-US but is longer (so you can see how your UI behaves in languages which take more space), and is different so you can see if you didn't hardcode any en-US string accidentally
* qps-plocm - Mirrored English. This pseudo-locale mimics how RTL language like Hebrew or Arabic might behave. You can test how your UI behaves in RTL scenarios.

Unfortunately, for this screen we relied in our tests on verifying that the time and date strings use pseudo-locale. With a switch from our DYI solution to Intl, we lost this ability because Intl doesn't localize the date or time string into accented or mirrored.

I had a choice of either removing the test, or manually running those two strings to morph them into a pseudo locale.
I chose the latter because I think that anyone who will want to switch to pseudo locale in Settings panel will want to see how it looks like on the date/time example.

In the future we will modify the code here to show 'ar' date/time for RTL pseudolocale to make it look even closer.

`navigator.mozL10n.qps` is just an Object that stores current pseudolocales, so I use it here to test if the language currently used is a pseudolocale.
It will not affect any production language behavior.